### ISSUE TITLE
EMI: various things appear as black silhouettes when using OpenGL

### ISSUE THREAD

**Issue by DavidGriffith** [residualvm/residualvm#1263]
When using plain OpenGL (no shaders), various objects appear as black silhouettes. The monkey that comes to greet Guybrush and Elaine is like this.  Then the catapult, pile of rocks, cactus, and basket of jerky pretzels.

**Comment by Stingm** [comment]
I just posted this same issue for pc version as well before seeing this. For me only software in non full screen mode works, I can't believe in all this time they haven't done anything about it. Hopefully  my post will get us some help.

**Comment by MrLint** [comment]
Well its still in there. Just discovered this myself today.

**Comment by Botje** [comment]
In order to be able to fix this, we need to reproduce it first.
If you can reproduce it consistently, please provide the following:
- a save game where it happens,
- the output of `apitrace trace` for a run where things appear black,
- information about your system (video card, opengl drivers)
- information about your game (if you can reproduce it on the demo as well, that would eliminate one variable)

**Comment by MrLint** [comment]
Save attached
Does not happen on SW rendering
trace attached
INFO: OpenGL Vendor: NVIDIA Corporation
INFO: OpenGL Renderer: NVS 5400M/PCIe/SSE2
INFO: OpenGL Version: 4.6.0 NVIDIA 388.75

GOG version of game. I'll have to get the demo and test. 

[efmi008.zip](https://github.com/residualvm/residualvm/files/2164678/efmi008.zip)


[residualvm.zip](https://github.com/residualvm/residualvm/files/2164641/residualvm.zip)

**Comment by Botje** [comment]
Thanks for the quick response, but the trace only goes into the main menu, not the game...

**Comment by MrLint** [comment]
Botje : Wow how odd. ok let me fire it up and do it again. I had the save loaded and moved around a bit.

**Comment by MrLint** [comment]
OK started the trace in GL mode, started at the bank manager and then rans around and ended up at brittany and her textures arent loaded either. However over character that are random walkers outside seem to be ok.

[residualvm.zip](https://github.com/residualvm/residualvm/files/2168336/residualvm.zip)
[efmi009.zip](https://github.com/residualvm/residualvm/files/2168338/efmi009.zip)

**Comment by MrLint** [comment]
Ok i think i know what happened. I specified the type of trace to gl and the trace file was now huge, even tho the defaults says gl. Whatever. Ok 2 things:

1) so when i first played I was using my laptop's intel gpu, i swapped it to the nvid to see if it would change anything. It didnt. However the trace here is on the intel , so its not merely nvid. 

2) so the fish monger is all black also, however when leaving that building the loading cd icon game up and suddenly.

The correct trace is 130M too big for the attachment. I had verbose on is that unnecessary for this purpose?

 the fish monger had his texture for a short moment before i was back out int the plaza

**Comment by Botje** [comment]
Yes, it is that big because it contains both the OpenGL command stream and any textures it uses.
If you can get that file uploaded somewhere, that would be great. If it helps with uploading, you can load the trace back into qapitrace and trim it so only the first few frames with black silhouettes are present.

**Comment by MrLint** [comment]
@Botje ok let see if this helps. https://www.dropbox.com/s/2r30s4fg0id88yq/residualvm.7z?dl=0

**Comment by Botje** [comment]
Alright! The trace indicates that the fish monger is being rendered with alpha 0 instead of 1! Now to figure out why ...

**Comment by MrLint** [comment]
Cool. I wonder if that explains the brief moment of render when the scene changed and it autosaved?

**Comment by Botje** [comment]
Can you try building and testing Botje/residualvm@b236a005 ? I see zero faces being rendered with alpha zero that way. I'm not sure if that is the proper fix thought...

**Comment by Botje** [comment]
Looked into all color maps: the majority just have alpha 1 everywhere, 25% have alpha 0 in them. Of those, a large majority have alpha 0 everywhere, with just a few that have both alpha 0 and alpha 1 bits.

**Comment by Botje** [comment]
And I've arrived at the final conclusion:
![snapshot01](https://user-images.githubusercontent.com/1329861/42655390-545e6ac4-861c-11e8-9a37-5b082d7a7470.png)
This is a rendering of Stan, according to the value of the color map.
The white, brown, and gold areas are places where the color map is 1, the hot pink is where the color map is 0. I believe these areas are normally covered by textures.

**Comment by MrLint** [comment]
ok please correct me here, but are you saying that if the color map is 0 it making it black instead of putting the texture on it? Or is it putting the texture on it and its back beccause the color is 0?

**Comment by MrLint** [comment]
Ahhhh, I think i understand.

**Comment by Botje** [comment]
I merged my pull request. 
@MrLint Can you check with a nightly build that includes this commit?

**Comment by MrLint** [comment]
It'd be my pleasure. Any thoughts on other EMI things we should dig into?:)
